sympy/solvers/ode.py

```diff
@@ -277,10 +277,10 @@
 #: ``best``, and ``all_Integral`` meta-hints should not be included in this
 #: list, but ``_best`` and ``_Integral`` hints should be included.
 allhints = (
-    "separable",
-    "1st_exact",
     "1st_linear",
     "Bernoulli",
+    "separable",
+    "1st_exact",
```

Comment:
@asmeurer I have corrected almost all the test cases except these two. 
Please help me in examining these test cases.
1. ```x,k,y,t=symbols('x C1 C2 k y t')
f=Function('f')
eq3 = (Eq(diff(x(t),t), y(t)*x(t)), Eq(diff(y(t),t), x(t)**3))
print dsolve(eq3)```
Without change:
```[x(t) == 6**(2/3)/(6*(-sinh(sqrt(C1)*(C2 + t)/2)/sqrt(C1))**(2/3)), y(t) == sqrt(C1 + C1/sinh(sqrt(C1)*(C2 + t)/2)**2)/3]```
With change:
Error trace: http://pastebin.com/Xid9dGt0

2. The details are here:
http://goo.gl/A26MwW

The detailed report of other test cases(including their runtime analysis) is here:
http://goo.gl/dDOYm0